apps/ejabberd/src/shaper.erl

```diff
@@ -1,89 +1,61 @@
-%%%----------------------------------------------------------------------
-%%% File    : shaper.erl
-%%% Author  : Alexey Shchepin <alexey@process-one.net>
-%%% Purpose : Functions to control connections traffic
-%%% Created :  9 Feb 2003 by Alexey Shchepin <alexey@process-one.net>
-%%%
-%%%
-%%% ejabberd, Copyright (C) 2002-2011   ProcessOne
-%%%
-%%% This program is free software; you can redistribute it and/or
-%%% modify it under the terms of the GNU General Public License as
-%%% published by the Free Software Foundation; either version 2 of the
-%%% License, or (at your option) any later version.
-%%%
-%%% This program is distributed in the hope that it will be useful,
-%%% but WITHOUT ANY WARRANTY; without even the implied warranty of
-%%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-%%% General Public License for more details.
-%%%
-%%% You should have received a copy of the GNU General Public License
-%%% along with this program; if not, write to the Free Software
-%%% Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
-%%% 02111-1307 USA
-%%%
-%%%----------------------------------------------------------------------
+%%==============================================================================
+%% Copyright 2016 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
 
 -module(shaper).
--author('alexey@process-one.net').
+-author('konrad.zemek@erlang-solutions.com').
 
--export([new/1, new1/1, update/2]).
+-export([new/1, update/2]).
 
--include("ejabberd.hrl").
+-record(shaper, {
+    max_rate :: undefined | pos_integer(),
+    tokens = 0 :: non_neg_integer(),
+    last_update = p1_time_compat:monotonic_time() :: integer()
+}).
 
--record(maxrate, {maxrate, lastrate, lasttime}).
-
--type shaper() :: none | #maxrate{}.
+-type shaper() :: #shaper{}.
 
 -export_type([shaper/0]).
 
 -spec new(atom()) -> shaper().
 new(Name) ->
-    Data = case ejabberd_config:get_global_option({shaper, Name, global}) of
-               undefined ->
-                   none;
-               D ->
-                   D
-           end,
-    new1(Data).
+    case ejabberd_config:get_global_option({shaper, Name, global}) of
+        undefined -> #shaper{};
+        none -> #shaper{};
```

Comment:
`none` is explicit setting for no shaper; it's there in the configs, I guess there's some value in that (e.g. in case we would want to have a different default?)